test/Constraints/diagnostics_swift4.swift

```diff
@@ -44,4 +44,14 @@ class R<K: Hashable, V> {
     var dict: [K:V] = [:]
     dict.forEach(body) // expected-error {{nested tuple parameter '(key: K, value: V)' of function '(((key: K, value: V)) throws -> Void) throws -> ()' does not support destructuring}}
   }
-}
\ No newline at end of file
+}
+
+// rdar://problem/32300339 - enum elements with labeled argument should be treated as functions when passed as value
```

Comment:
Please put this in a new file, test/Constraints/enum_cases.swift.

Both files need a lot more tests, for example it would be nice to test:

- case has no label
- case has label
- case is a tuple with multiple elements, some of which have labels
- assign case to a 'let' binding with an explicit type